### DISCUSSION TITLE
安全问题思考

### DISCUSSION THREAD

**Discussion by BrandonStudio** [lobehub/lobehub#1915] [General]
目前模型和插件调用全部通过后端发送，这意味着客户端设置的各种 API key 会发送到后端。只要对后端稍作改动，就可以记录全部私密信息。

但是这个问题好像有点无解，即使添加警告也可以被部署者删除

**Comment by arvinxx** [comment]
可以看看 https://github.com/lobehub/lobe-chat/discussions/728 未来会做client端方案的

**Reply by arvinxx** [reply]
目前已经支持 client 端请求了，用户可以自行决策是否使用 client 端请求。

**Comment by BrandonStudio** [comment]
另外关于使用数据库存储同步信息的功能，应当完全由用户决定存储位置（由用户填写数据库连接，如连接字符串或redis链接），否则全部对话（可能也包括配置）将对存储提供者（即部署者）可见。

**Reply by cy948** [reply]
当前的 sync 是 webRTC 方式，数据不经过中心服务器。

**Reply by BrandonStudio** [reply]
我说的是未来

**Reply by cy948** [reply]
> 我说的是未来

那肯定是由用户指定的。

**Reply by arvinxx** [reply]
使用数据库方案肯定是用户自己部署了。或者用我们提供的 cloud
